components/main/GlobalAlert.tsx

```diff
@@ -0,0 +1,42 @@
+import { NavLink, useLocation } from "react-router-dom";
+import { useAppSelector } from "../../redux/hooks/hooks";
+
+const GlobalAlert = () => {
+  return (
+    <div
+      className="app-global-alert"
+      id="app-global-alert"
+      role="complementary"
+    >
+      <div className="nhsuk-width-container">
+        <CojAlert />

```

Comment:
The `GlobalAlert` is always rendered, the `CojAlert` may or may not be rendered based on the current state. If no alerts are rendered then GlobalAlert will still exist in the DOM but not be displayed.

The decision to render `CojAlert` was deferred to itself rather than logic here so that we could avoid complicating the GlobalAlert logic if/when further types of alerts are added.

Reply:
Yeah , I see your thinking here to make the GlobalAlert more generic. Maybe we could conditionally render it by e.g. checking an array of functions e.g. showCojAlert, showNextThingAlert etc.., if any are true then GlobalAlert is rendered along with any child alerts that were true?